### DISCUSSION TITLE
Clarify "Filter by trace ID" and "Filter by span ID" toggles

### DISCUSSION THREAD

**Discussion by ptodev** [grafana/grafana#72262] [Observability]
When configuring [traces to logs](https://grafana.com/docs/grafana/latest/explore/trace-integration/#trace-to-logs), there are [toggles](https://grafana.com/docs/grafana/latest/datasources/tempo/#trace-to-logs) for "Filter by trace ID" and "Filter by span ID". It would be good to clarify how this filtering is done and if the user has to do any extra steps to make it work. For example, is it sufficient to have the trace ID in the log body or should there be a Loki label for it?

This was brought up in a [conversation](https://grafana.slack.com/archives/C01050C3D8F/p1690281419187399?thread_ts=1690213243.858029&cid=C01050C3D8F) with a customer.

**Comment by aocenas** [comment]
@ptodev not exactly sure if I understand the question. Tempo part is just to create the query for loki. We could probably show some sort of example query in the config page to illustrate that a bit. We basically just append `|="${__span.traceId}"` to the query where the `${__span.traceId}` will be replaces by the actual traceId from the trace. So would a query example in the config page help?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). If you can clarify what it is you're looking for, feel free to make an [issue](https://github.com/grafana/grafana/issues) and we can go from there.

**Comment by iby-dev** [comment]
Hi, I would like this functionality clarifying too, the OP question is very clear to me. It matches my scenario too. It is not clear precisely how the matching logic works given the generated query. i,.e when you try Filter by Span Id, i get bupkiss, nada, nothing. So a clear explaination of how these two toggles work would go along way in improving the user experience of navigating very long traces.
